syftlib/src/main/java/org/openmined/kotlinsyft/WebRTC.kt

```diff
@@ -0,0 +1,304 @@
+            Log.e(TAG, "error creating : $error")
+        }
+
+    }
+
+    inner class PeerConnectionObserver(
+        private val worker_id: String,
+        private val creatorType: SDP_type
+    ) : Observer {
+
+        override fun onIceCandidate(new_candidate: IceCandidate?) {
+            if (new_candidate != null) {
+                Log.d(TAG, "Saving new ICE Candidate")
+                peers[worker_id]?.candidateQueue?.add(new_candidate)
+            }
+        }
+
+        override fun onDataChannel(dc: DataChannel) {
+            Log.d(TAG, "Calling onDataChannel ${dc.label()}")
+            dc.registerObserver(DataChannelObserver(dc))
+            peers[worker_id]?.channel = dc
+
+        }
+
+        override fun onIceConnectionReceivingChange(p0: Boolean) {
+            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+        }
+
+        override fun onIceConnectionChange(p0: IceConnectionState?) {
+            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+        }
+
+        override fun onIceGatheringChange(gatheringState: IceGatheringState) {}
+
+        override fun onSignalingChange(signalingState: SignalingState) {
+            Log.d(TAG, "Signalling State: $signalingState")
+        }
+
+        override fun onIceCandidatesRemoved(p0: Array<out IceCandidate>?) {
+            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+        }
+
+        override fun onRenegotiationNeeded() {
+            TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+        }
+
+        override fun onAddTrack(p0: RtpReceiver?, p1: Array<out MediaStream>?) {}
+        override fun onRemoveStream(p0: MediaStream?) {}
+        override fun onAddStream(p0: MediaStream?) {}
+
+    }
+
+    class DataChannelObserver(private val dataChannel: DataChannel?) : DataChannel.Observer {
+        override fun onMessage(buffer: DataChannel.Buffer) {
+            if (buffer.binary) {
+                Log.d(TAG, "Data channel received binary message at $dataChannel")
+                return
+            }
+
+            Log.d(TAG, "Data channel message ${buffer.data}")
+        }
+
+        override fun onBufferedAmountChange(previous_amount: Long) {
+            Log.d(
+                TAG,
+                "buffered amount changed from $previous_amount to ${dataChannel?.bufferedAmount()}"
+            )
+        }
+
+        override fun onStateChange() {
+            Log.d(TAG, "Data channel state changed to ${dataChannel?.state()}")
+        }
+
+    }
+
+    class Peer(
+        var connection: PeerConnection?,
+        var channel: DataChannel?,
+        val peerConnectionObserver: PeerConnectionObserver,
+        val sdpObserver: SDPObserver
+    ) {
+        val candidateQueue = ArrayList<IceCandidate>()
```

Comment:
Maybe a `MutableList`?

Reply:
Aren't they same?

Reply:
It's good practice since the mutable list does not define the implementation is using (though today ArrayList).
Not a big deal though

Reply:
Aren't they same? Mutable list, on the other hand, is susceptible to  changes by Kotlin